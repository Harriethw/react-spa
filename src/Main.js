import React, { Component } from "react";
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import Card from 'react-bootstrap/Card';
import {
    Route,
    NavLink,
    HashRouter
} from 'react-router-dom';
import Schedule from "./Schedule";
import Location from "./Location";
import FAQ from "./FAQ";

class Main extends Component {
    render() {
        return (
            <HashRouter>
                <div>
                    <Navbar bg="light" expand="lg">
                        <Navbar.Brand href="/">Away Day 2019</Navbar.Brand>
                        <Navbar.Toggle aria-controls="basic-navbar-nav" />
                        <Navbar.Collapse id="basic-navbar-nav">
                            <Nav className="mr-auto">
                                <Nav.Link><NavLink to="/">Schedule</NavLink></Nav.Link>
                                <Nav.Link><NavLink to="/location">Location</NavLink></Nav.Link>
                                <Nav.Link><NavLink to="/faq">FAQ</NavLink></Nav.Link>
                            </Nav>
                        </Navbar.Collapse>
                    </Navbar>
                    <Card className="content" body>
                        <Route exact path="/" component={Schedule} />
                        <Route path="/location" component={Location} />
                        <Route path="/faq" component={FAQ} />
                    </Card>
                </div>
            </HashRouter>
        );
    }
}

export default Main;

