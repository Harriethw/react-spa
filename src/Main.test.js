import React from 'react';
import { shallow, mount, simulate } from 'enzyme';
import Main from './Main';
import { MemoryRouter } from 'react-router-dom';

describe('Main', () => {
  it('should render correctly ', () => {
    const component = shallow(<Main />);
  
    expect(component).toMatchSnapshot();
  });

  it('should link to FAQs', () => {
    const component = mount(<MemoryRouter>
            <Main />
    </MemoryRouter>);
    
    component.find('a').last().simulate('click', { button: 0 });
    component.containsMatchingElement('<h2>GOT QUESTIONS?</h2>');
    component.unmount();
  })
});