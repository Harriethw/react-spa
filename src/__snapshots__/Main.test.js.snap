// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly  1`] = `
<HashRouter>
  <div>
    <Bootstrap(undefined)
      bg="light"
      expand="lg"
    >
      <NavbarBrand>
        Away Day 2019
      </NavbarBrand>
      <NavbarToggle
        aria-controls="basic-navbar-nav"
        label="Toggle navigation"
      />
      <NavbarCollapse
        id="basic-navbar-nav"
      >
        <Nav
          className="mr-auto"
          fill={false}
          justify={false}
        >
          <NavLink
            as={[Function]}
            disabled={false}
          >
            <NavLink
              to="/"
            >
              Schedule
            </NavLink>
          </NavLink>
          <NavLink
            as={[Function]}
            disabled={false}
          >
            <NavLink
              to="/location"
            >
              Location
            </NavLink>
          </NavLink>
          <NavLink
            as={[Function]}
            disabled={false}
          >
            <NavLink
              to="/faq"
            >
              FAQ
            </NavLink>
          </NavLink>
        </Nav>
      </NavbarCollapse>
    </Bootstrap(undefined)>
    <Card
      body={true}
      className="content"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/location"
      />
      <Route
        component={[Function]}
        path="/faq"
      />
    </Card>
  </div>
</HashRouter>
`;
